packages/x-date-pickers-pro/src/DateRangePickerDay/DateRangePickerDay.tsx

```diff
@@ -303,44 +260,66 @@ const DateRangePickerDayRaw = React.forwardRef(function DateRangePickerDay(
   const props = useThemeProps({ props: inProps, name: 'MuiDateRangePickerDay' });
   const {
     className,
-    day,
-    outsideCurrentMonth,
+    classes: classesProp,
     isEndOfHighlighting,
     isEndOfPreviewing,
     isHighlighting,
     isPreviewing,
     isStartOfHighlighting,
     isStartOfPreviewing,
-    selected = false,
     isVisuallySelected,
     sx,
     draggable,
     isFirstVisibleCell,
     isLastVisibleCell,
+    day,
+    selected,
+    disabled,
+    today,
+    outsideCurrentMonth,
+    disableMargin,
+    disableHighlightToday,
+    showDaysOutsideCurrentMonth,
     ...other
   } = props;
 
   useLicenseVerifier('x-date-pickers-pro', releaseInfo);
   const utils = useUtils();
 
-  const isEndOfMonth = utils.isSameDay(day, utils.endOfMonth(day));
-  const isStartOfMonth = utils.isSameDay(day, utils.startOfMonth(day));
-
   const shouldRenderHighlight = isHighlighting && !outsideCurrentMonth;
   const shouldRenderPreview = isPreviewing && !outsideCurrentMonth;
 
-  const ownerState = {
-    ...props,
+  const pickersDayOwnerState = usePickerDayOwnerState({
+    day,
     selected,
-    isStartOfMonth,
-    isEndOfMonth,
-    draggable,
-    isFirstVisibleCell,
-    isLastVisibleCell,
-    isHiddenDayFiller: outsideCurrentMonth && !other.showDaysOutsideCurrentMonth,
+    disabled,
+    today,
+    outsideCurrentMonth,
+    disableMargin,
+    disableHighlightToday,
+    showDaysOutsideCurrentMonth,
+  });
+
+  const ownerState: DateRangePickerDayOwnerState = {
```

Comment:
I think  this covers almost everything that I use for the styling. 🤔 I also added `firstDayOfWeek` and `lastDayOfWeek` to the `ownerState` for styling, and those are the only ones that I think are missing from here 🤔 Do you think we could add them, or should we try to find a workaround?
![image](https://github.com/user-attachments/assets/c93cef1b-1f00-436a-a3c0-0d2d5f0fae07)

Reply:
> I also added firstDayOfWeek and lastDayOfWeek to the ownerState for stylin

I can add those as well, they are present in the Base UI component anyway.

Reply:
That would be great!

Reply:
Added :ok_hand: